cpp/src/io/utilities/data_sink.cpp

```diff
@@ -38,6 +38,8 @@ class file_sink : public data_sink {
 
     if (detail::cufile_integration::is_kvikio_enabled()) {
       _kvikio_file = kvikio::FileHandle(filepath, "w");
+      CUDF_LOG_INFO("Writing a file using kvikIO, with compatibility mode {}.",
+                    _kvikio_file.is_compat_mode_on() ? "on" : "off");
```

Comment:
Should we also log for the `else` branch?

Reply:
No need to log anything in the else branch because it calls `make_cufile_output`, which includes the logging for that path (whether GDS was used or not).
Same goes for the other file, just with `make_cufile_input`.